app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaApp.kt

```diff
@@ -31,93 +31,154 @@ import androidx.compose.material3.ExperimentalMaterial3Api
+import com.google.samples.apps.nowinandroid.feature.settings.SettingsDialog
 import com.google.samples.apps.nowinandroid.navigation.NiaNavHost
 import com.google.samples.apps.nowinandroid.navigation.TopLevelDestination
 
 @OptIn(
     ExperimentalMaterial3Api::class,
     ExperimentalLayoutApi::class,
-    ExperimentalComposeUiApi::class
+    ExperimentalComposeUiApi::class,
+    ExperimentalLifecycleComposeApi::class
 )
 @Composable
 fun NiaApp(
     windowSizeClass: WindowSizeClass,
-    appState: NiaAppState = rememberNiaAppState(windowSizeClass)
+    networkMonitor: NetworkMonitor,
+    appState: NiaAppState = rememberNiaAppState(
+        networkMonitor = networkMonitor,
+        windowSizeClass = windowSizeClass
+    ),
 ) {
-    NiaTheme {
-        val background: @Composable (@Composable () -> Unit) -> Unit =
-            when (appState.currentDestination?.route) {
-                TopLevelDestination.FOR_YOU.name -> { content ->
-                    NiaGradientBackground(content = content)
-                }
-                else -> { content -> NiaBackground(content = content) }
+    val background: @Composable (@Composable () -> Unit) -> Unit =
+        when (appState.currentDestination?.route) {
+            TopLevelDestination.FOR_YOU.name -> {
+                content ->
+                NiaGradientBackground(content = content)
             }
+            else -> { content -> NiaBackground(content = content) }
+        }
 
-        background {
-            Scaffold(
-                modifier = Modifier.semantics {
-                    testTagsAsResourceId = true
-                },
-                containerColor = Color.Transparent,
-                contentColor = MaterialTheme.colorScheme.onBackground,
-                contentWindowInsets = WindowInsets(0, 0, 0, 0),
-                bottomBar = {
-                    if (appState.shouldShowBottomBar) {
-                        NiaBottomBar(
-                            destinations = appState.topLevelDestinations,
-                            onNavigateToDestination = appState::navigateToTopLevelDestination,
-                            currentDestination = appState.currentDestination
-                        )
-                    }
+    background {
+
+        val snackbarHostState = remember { SnackbarHostState() }
+
+        Scaffold(
+            modifier = Modifier.semantics {
+                testTagsAsResourceId = true
+            },
+            containerColor = Color.Transparent,
+            contentColor = MaterialTheme.colorScheme.onBackground,
+            contentWindowInsets = WindowInsets(0, 0, 0, 0),
+            snackbarHost = { SnackbarHost(snackbarHostState) },
+            topBar = {
+                // Show the top app bar on top level destinations.
+                val destination = appState.currentTopLevelDestination
+                if (destination != null) {
+                    NiaTopAppBar(
```

Comment:
A side-effect of moving the top app bar to the outermost scope: The top app bar is now rendered over the rest of the content in the scaffold. This means that the top app bar is overlapping with the nav rail, and blocks tapping on the top-most navigation rail item.

I think `Modifier.zIndex` would be the most straightforward workaround?

Reply:
Constantly amazed at how you spot things like this. Added the suggested workaround. 

Also, we should really add a large screen device to our CI workflow. https://github.com/android/nowinandroid/issues/353

Reply:
Paketler:
WhatsApp: 2.23.8.76
Haritalar: 11.76.0300
CarStream: 2.0.5
YouTube Music: 5.54.52
Fermata Control: 1.8.11
Google Play Store: 35.4.10-21 [0] [PR] 524351809
Google Play hizmetleri: 23.15.56 (150400-527597497)
Fermata Auto: 1.9.1
Mesajlar: messages.android_20230329_00_RC03.phone_dynamic
Google: 14.16.27.28.arm64
USB Audio Player PRO: 6.1.0.4
fizy: 9.1.7
Signal: 6.18.4
Android Auto: 9.3.631434-release
Telegram: 9.6.3
Telegram: 9.6.2
Messenger: 406.0.0.13.115
Google Play Kitaplar: 2023.4.3.0.4
CarStream: 1.8.3

İzinler:
phone: açık
contacts: açık
locatıon: açık
sms: açık
mıcrophone: açık
calendar: açık
call_log: açık
nearby_devıces: açık